pkg/apis/pipeline/v1alpha1/taskrun_defaults.go

```diff
@@ -29,11 +29,22 @@ func (tr *TaskRun) SetDefaults(ctx context.Context) {
 }
 
 func (trs *TaskRunSpec) SetDefaults(ctx context.Context) {
+	cfg := config.FromContextOrDefaults(ctx)
 	if trs.TaskRef != nil && trs.TaskRef.Kind == "" {
 		trs.TaskRef.Kind = NamespacedTaskKind
 	}
-	cfg := config.FromContextOrDefaults(ctx)
+
+	var timeout *metav1.Duration
+	if IsUpgradeViaDefaulting(ctx) {
+		// This case is for preexisting `TaskRun` before 0.5.0, so let's
+		// add the old default timeout.
+		// Most likely those TaskRun passing here are already done and/or already running
+		// and were meant to have the 10 minutes default timeout
+		timeout = &metav1.Duration{Duration: 10 * time.Minute}
```

Comment:
I hard-coded *old* value. We could just use `config.DefaultTimeoutMinutes` here instead.

Reply:
I would rather use the new value - seems odd to sometimes default to old value, sometimes default to the new value